staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -349,6 +349,16 @@ type ListOptions struct {
 	// add, update, and remove notifications. Specify resourceVersion.
 	// +optional
 	Watch bool `json:"watch,omitempty" protobuf:"varint,3,opt,name=watch"`
+	// allowBookmarks requests watch events with type "BOOKMARK".
+	// Servers that do not implement bookmarks may ignore this flag and
+	// bookmarks are sent at the server's discretion. Clients should not
+	// assume bookmarks are returned at any specific interval, not may they
+	// assume the server will send any BOOKMARK event during a session.
+	// If this is not a watch, this field is ignored.
+	// If the feature gate WatchBookmarks is not enabled, this field is ignored.
+	// +optional
+	AllowBookmarks bool `json:"allowBookmarks,omitempty"`
```

Comment:
if this is watch-specific, should it be reflected in the name? `allowWatchBookmarks`? `allowWatchBookmarkEvents`?

Reply:
I like it - changed to allowWatchBookmarks